src/main.rs

```diff
@@ -77,7 +77,11 @@ fn main() -> Result<()> {
 
     let ctrlc = Arc::new(AtomicBool::new(false));
     // TODO: make this conditional in the future
-    ctrlc_protection(&mut engine_state, &ctrlc);
+    // This block sets up Ctrl-C signal handling to safely handle user interrupts.
+    // Consider making this configurable in future versions.
+    if cfg!(feature = "ctrlc-protection") {
```

Comment:
I don't see this featrure in nushell. What am I missing?

Reply:
It was a TODO in main.rs. I thought it must be done in some way.

Reply:
I don't see this feature so landing this PR would effectively disable ctrlc_protection right now.